cg/services/run_devices/abstract_classes.py

```diff
@@ -83,6 +86,23 @@ def post_process(self, run_name: str, dry_run: bool = False):
         """Store sequencing metrics in StatusDB and relevant files in Housekeeper."""
         pass
 
+    def post_process_all(self, dry_run: bool = False) -> bool:
+        """Post-process all PacBio runs in the sequencing directory."""
+        LOG.info("Starting PacBio post-processing for all runs")
+        is_post_processing_successful = True
+        for run_name in self._get_all_run_names():
+            try:
+                self.post_process(run_name=run_name, dry_run=dry_run)
+            except Exception as error:
+                LOG.error(f"Could not post-process {run_name}: {error}")
+                is_post_processing_successful = False
+        return is_post_processing_successful
```

Comment:
Can we make this raise an error instead of swapping a bool when something goes wrong? We can then except this specific error in the CLI and handle it there

Reply:
The problem I see with raising an error here is it would terminate the iteration early and some smart cells would not start their post-processing despite being able to

Reply:
For me the most reasonable would probably be to raise an error in post_process_run which post_process_all catches. And if any run has crashed in the post_process_all, then it can raise an error which we can catch in the CLI. It does sound like the CLI might be doing a bit too much at that stage though..

Reply:
> The problem I see with raising an error here is it would terminate the iteration early and some smart cells would not start their post-processing despite being able to

Yes, the post_process_all function makes things more complicated that just iterating through the dirs and running the post process function in the CLI. I think that would prevent the need for all this extra stuff.

Should we re think the approach?

Reply:
To me the tradeoff would still be that we would need to go

Fetch device from directory name -> Fetch expected directory structure from device -> get run name -> from run name get device (which we in principle already have determined and uses a regex) -> post process run

To me, handling errors in the current approach is still more attractive and simpler, but we could certainly go for another refinement.

Reply:
closing after refinement